File: the_ultimate_square/ultimate.c
#include <stdio.h>

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int n;
        scanf("%d", &n);
        if (n % 2)
        {
            printf("%d\n", n / 2 + 1);
        }

        else
        {
            printf("%d\n", n / 2);
        }
    }
}

File: the_ultimate_square/ultimate.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        n = int(input())
        if n % 2:
            print(n // 2 + 1)
        else:
            print(n // 2)
    
    
if __name__ == "__main__":
    main()
